### PR TITLE
[16.0][MIG] - sale_triple_discount

### PR DESCRIPTION
**Pull request by sbejaoui**

### CODE REVIEW

**Review by legalsylvain — COMMENTED** [review]
Hi. Thanks for porting this module !
- could you take a look on red CI ?
- could you cherry pick https://github.com/OCA/sale-workflow/pull/2101 if you think it's relevant ?

Thanks !

**Comment by sbejaoui** [comment]
> Hi. Thanks for porting this module !
> 
> * could you take a look on red CI ?
> * could you cherry pick #2101 if you think it's relevant ?
> 
> Thanks !

CI will be fixed once https://github.com/OCA/account-invoicing/pull/1269 merged.

File: sale_triple_discount/tests/test_sale_triple_discount.py
Comment (lmignon) [inline sale_triple_discount/tests/test_sale_triple_discount.py]: Why do you change from `setUpClass` to `setUp`  It's better to create all the required stuff into setUpclass than in stUp since setUpClass is executed only once for all the test (with a rollback to savepoint after each test) bu setUp is executed bedore each test...

**Review by lmignon — CHANGES_REQUESTED** [review]

**Comment by lmignon** [comment]
@sbejaoui You should add a test-requirements.txt file into a separate commit with 'DO NOT MERGE' as comment to to install `account_invoice_triple_discount`  from https://github.com/OCA/account-invoicing/pull/1269. That will allow to run tests and make functional tests on runboat.

**Review by lmignon — APPROVED** [review]
Thank you @sbejaoui LGTM (Code review + functional tests)

**Review by marielejeune — APPROVED** [review]
LGTM (Code review + functional tests)

**Comment by lmignon** [comment]
@sbejaoui https://github.com/OCA/account-invoicing/pull/1269 is merged. Can you remove the test dependency PLZ so we can move forward.

**Comment by rousseldenis** [comment]
@legalsylvain Could you update your review ?

**Comment by rousseldenis** [comment]
@sbejaoui Could you update this as https://github.com/OCA/account-invoicing/pull/1269 is merged ?

**Comment by sbejaoui** [comment]
> @sbejaoui Could you update this as [OCA/account-invoicing#1269](https://github.com/OCA/account-invoicing/pull/1269) is merged ?

done

**Comment by lmignon** [comment]
@OCA/crm-sales-marketing-maintainers Someone to merge this one? Thank!

**Comment by rousseldenis** [comment]
@legalsylvain It seems the PR you point out is referring to code that does not exist here (no sale order override)

**Review by rousseldenis — APPROVED** [review]

**Comment by rousseldenis** [comment]
So, let's go merge and make improvements later if needed.

/ocabot merge patch

**Pull request merged**
